### PR TITLE
several podman logs fixes

### PR DESCRIPTION
**Pull request by Luap99**
see individual commits for details, I recommend you review this commit by commit and not the full diff at once.

#### Does this PR introduce a user-facing change?

```release-note
Fixed a bug where podman logs --since --follow would not follow and just exit with the journald driver.
Fixed a bug where podman logs --until --follow would not exit after the given until time.
The journald driver for both events and logs is now more efficient when --since is used, it will now seek directly to the correct time instead of reading all entries from he journal.
```

### CODE REVIEW

**Comment by Luap99** [comment]
@edsantiago PTAL at the tests

File: libpod/container_log.go
Comment (rhatdan) [inline libpod/container_log.go:76]: If options.Until ==NIL, does this sleep forever?

BTW The commit message here is badly spelled.

Comment (Luap99) [reply libpod/container_log.go:76]: good catch, it also needs to ignore the case where until is the past

**Comment by rhatdan** [comment]
LGTM
Nice work.
@containers/podman-maintainers PTAL

File: test/system/035-logs.bats
Comment (edsantiago) [inline test/system/035-logs.bats:340]: These two tests pass on `main`.

File: test/system/035-logs.bats
Comment (edsantiago) [inline test/system/035-logs.bats:358]: Prefixing with `PODMAN_TIMEOUT=10` might be friendly here, since the test should never take longer than that.

**Review by edsantiago — COMMENTED** [review]
The two new `since` tests pass on `main`. If you believe that your PR fixes some behavior in `--since`, you may want to reevaluate your tests so they fail on main.

The copypaste duplication squicks me out. I can't find an obvious way to clean it up, but I'm going to keep looking. If you can think of a way to refactor, I'd appreciate it.

File: test/system/035-logs.bats
Comment (Luap99) [reply test/system/035-logs.bats:340]: oh yeah you are right, I have to add a `sleep 1` before ` PODMAN_TIMEOUT=2 run_podman 124 ...` to make it fail. The logs command need to wait for the amount specified with --since in order to reproduce. I think I can try to use something like 0.5 seconds to make CI a bit faster.

Comment (Luap99) [reply test/system/035-logs.bats:340]: also this only effects the journald backend, the k8s-file should already work but I like to test both regardless

File: test/system/035-logs.bats
Comment (edsantiago) [inline test/system/035-logs.bats:368]: Very minor, not worth fixing unless you need to re-push: the word is **too** (t-o-o, two "o"s).

File: libpod/container_log_linux.go
Comment (edsantiago) [inline libpod/container_log_linux.go:99]: I don't understand this sentence, but will not block over it. I will trust that more knowledgeable people will either understand it, or block the PR requesting clarification.

**Review by edsantiago — COMMENTED** [review]
Minor language nits. LGTM.

File: libpod/container_log_linux.go
Comment (vrothberg) [inline libpod/container_log_linux.go:95]: Please unlock earlier.  The lock is only needed for the state check right below.  Just to be defensive for future changes which may perform more work after.

File: test/system/035-logs.bats
Comment (vrothberg) [inline test/system/035-logs.bats:308]: Double blank line.

File: libpod/container_log_linux.go
Comment (vrothberg) [reply libpod/container_log_linux.go:99]: The "not being in the journal" part confused me as well but the test gave clarity.  It's probably worth mentioning that `--since` may be set after the start event, so we need to take the initial container state into account before reading the journal.

File: libpod/container_log_linux.go
Comment (vrothberg) [inline libpod/container_log_linux.go:88]: Very nice improvement!

**Review by vrothberg — COMMENTED** [review]
LGTM, only minor nits.

Nice idea to share the code!

File: libpod/container_log_linux.go
Comment (Luap99) [reply libpod/container_log_linux.go:99]: --since just triggers it more easily but it is actually a general problem, if your container runs long enough the journal will be rotated at some point and you can loose the container start event.

File: libpod/container_log_linux.go
Comment (Luap99) [reply libpod/container_log_linux.go]: fixed

File: libpod/container_log_linux.go
Comment (Luap99) [reply libpod/container_log_linux.go]: I made it more clear when/how it can happen.

**Review by vrothberg — APPROVED** [review]
LGTM, nice work!

**Pull request merged**
